src/PrestaShopBundle/Security/Admin/EmployeeProvider.php

```diff
@@ -121,6 +131,6 @@ public function refreshUser(UserInterface $employee)
      */
     public function supportsClass($class)
     {
-        return $class === 'PrestaShopBundle\Security\Admin\Employee';
+        return $class === Employee::class;
```

Comment:
I know it's now deprecated but shouldn't this use `instanceof` instead?

Reply:
depends on if you want to test inheritance or not :thinking:

Reply:
it's tightly coupled to the legacy `Employee` class so I guess it's not good to test inheritance :slightly_smiling_face:

Reply:
@jolelievre This is the contrary: the current test checks that given class is **exactly** an Employee class, and not a children of it. Rokas want sot make sure given class is original Employee class because a child of this class might have a different structure that would break the authentication system.

@rokaszygmantas but ... you test here this is an instance of `PrestaShopBundle\Security\Admin\Employee`, not legacy Employee class ?

Reply:
@matks yes, because `PrestaShopBundle\Security\Admin\Employee` is coupled to legacy `Employee` - you must inject `Employee` in the constructor of `PrestaShopBundle\Security\Admin\Employee` otherwise everything will crash